File: iosApp/iosApp/view/touchdown/model/Product.swift
//
//  Product.swift
//  Coca
//
//  Created by Huynh Phong on 04/04/2022.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import SwiftUI

struct Product: Codable, Identifiable {
    let id: Int
    let name: String
    let image: String
    let price: Int
    let description: String
    let color: [Double]
}

extension Product {
    var formattedPrice: String {
        return "$\(price)"
    }
    var colorValue: Color {
        return Color(red: color[0], green: color[1], blue: color[2])
    }
}

File: shared/src/commonMain/kotlin/com/wind/book/model/touchdown/Product.kt
package com.wind.book.model.touchdown

import com.wind.book.Parcelable
import com.wind.book.Parcelize
import kotlinx.serialization.Serializable

@Serializable
@Parcelize
data class Product(
    val id: Int,
    val name: String,
    val image: String,
    val price: Int,
    val description: String,
    val color: List<Double>
) : Parcelable {

    fun formattedPrice() = "$$price"
}
